cloudsql/sqlserver/database-sql/cloudsql_test.go

```diff
@@ -22,19 +22,54 @@ import (
 	"testing"
 )
 
+type commonTestInfo struct {
+	dbName                 string
+	dbPass                 string
+	dbUser                 string
+	dbPort                 string
+	instanceConnectionName string
+}
+
+func getCommonTestInfo() commonTestInfo {
+	return commonTestInfo{
+		dbName:                 os.Getenv("SQLSERVER_DATABASE"),
+		dbPass:                 os.Getenv("SQLSERVER_PASSWORD"),
+		dbPort:                 os.Getenv("SQLSERVER_PORT"),
+		dbUser:                 os.Getenv("SQLSERVER_USER"),
+		instanceConnectionName: os.Getenv("SQLSERVER_INSTANCE"),
+	}
+}
+
 func TestIndex(t *testing.T) {
 	if os.Getenv("GOLANG_SAMPLES_E2E_TEST") == "" {
 		t.Skip()
 	}
+
+	_commonTestInfo := getCommonTestInfo()
+
 	tests := []struct {
 		dbHost string
+		common commonTestInfo
 	}{
-		{dbHost: os.Getenv("DB_HOST")},
+		{dbHost: os.Getenv("SQLSERVER_HOST"), common: _commonTestInfo},
 	}
 
+	// Capture original values
+	oldDBHost := os.Getenv("DB_HOST")
+	oldDBName := os.Getenv("DB_NAME")
+	oldDBPass := os.Getenv("DB_PASS")
+	oldDBPort := os.Getenv("DB_PORT")
+	oldDBUser := os.Getenv("DB_USER")
+	oldInstance := os.Getenv("INSTANCE_CONNECTION_NAME")
+
 	for _, test := range tests {
-		oldDBHost := os.Getenv("DB_HOST")
+		// Set overwrites
 		os.Setenv("DB_HOST", test.dbHost)
```

Comment:
In general, mutuating environment variables in tests makes it _really_ difficult to parallelize tests and leads to flakes. What is the value in calling `Setenv` here? What problem are we trying to solve?

Reply:
These are integration tests, so we're trying to run the code in for-real mode to make real connections to the database, and that code reads environment variables.

Reply:
As per our offline conversation, these environment variable mutations will land for now, as e2e tests are run less frequently. We'll rethink the issue if problems surface.